types/record.d.ts

```diff
@@ -19,6 +19,8 @@
 
 declare type Visitor = (value: any, key: string, record: Record) => void
 
+declare type MapVisitor = (value: any, key: string, record: Record) => any
```

Comment:
I think it's better to use a generic type here instead of `any`.

Visitor type would be:

```typescript
declare type MapVisitor<T> = (value: any, key: string, record: Record) => T
```

and `map` function would become:

```typescript
map<T>(visitor: MapVisitor<T>): T[];
```